probe-rs/src/architecture/arm/sequences/cc13xx_cc26xx.rs

```diff
@@ -0,0 +1,340 @@
+use crate::architecture::arm::ArmError;
+
+/// Marker struct indicating initialization sequencing for cc13xx_cc26xx family parts.
+pub struct CC13xxCC26xx {}
+
+// IR register values, see https://www.ti.com/lit/ug/swcu185f/swcu185f.pdf table 6-7
+const IR_ROUTER: u64 = 0x02;
+const IR_CONNECT: u64 = 0x07;
+const IR_BYPASS: u64 = 0x3F;
+const IR_LEN_IN_BITS: u8 = 6;
+
+#[derive(PartialEq)]
+enum JtagState {
+    RunTestIdle = 0x1,
+    SelectDRScan = 0x2,
+    ShiftDR = 0x03,
+    ShiftIR = 0x04,
+}
+
+// Set the bottom n bits of a u64 to 1
+// This is lifted directly from:
+// https://users.rust-lang.org/t/how-to-make-an-integer-with-n-bits-set-without-overflow/63078/6
+fn set_n_bits(x: u32) -> u64 {
+    u64::checked_shl(1, x).unwrap_or(0).wrapping_sub(1)
+}
+
+impl CC13xxCC26xx {
+    /// Create the sequencer for the cc13xx_cc26xx family of parts.
+    pub fn create() -> Arc<Self> {
+        Arc::new(Self {})
+    }
+
+    /// This function implements a Zero Bit Scan(ZBS)
+    ///
+    /// The ZBS defined in section 6.2.2.1 of this document:
+    /// https://www.ti.com/lit/ug/swcu185f/swcu185f.pdf
+    ///
+    /// This function assumes that the JTAG state machine is in the Run-Test/Idle state
+    ///
+    /// * `interface` - Reference to interface to interact with CmsisDap
+    fn zero_bit_scan(&self, interface: &mut dyn DapProbe) -> Result<(), ArmError> {
+        // Enter DRSELECT state
+        interface.jtag_sequence(1, true, 0x01)?;
+        // Enter DRCAPTURE state
+        interface.jtag_sequence(1, false, 0x01)?;
+        // Enter DREXIT1 state
+        interface.jtag_sequence(1, true, 0x01)?;
+        // Enter DRPAUSE state
+        interface.jtag_sequence(1, false, 0x01)?;
+        // Enter DREXIT2 state
+        interface.jtag_sequence(1, true, 0x01)?;
+        // Enter DRUPDATE state
+        interface.jtag_sequence(1, true, 0x01)?;
+        // Enter Run/Idle state
+        interface.jtag_sequence(1, false, 0x01)?;
+
+        Ok(())
+    }
+    /// Load a value into the IR register
+    ///
+    /// This function is a wrapper on `shift_reg` that loads a value into the IR register
+    ///
+    /// * `interface` - Reference to interface to interact with CmsisDap
+    /// * `cycles`    - Number of TCK cycles to shift in the data to IR
+    /// * `reg`       - The value to shift into either IR
+    /// * `state`     - The current state of the JTAG state machine. Note this will be updated by this function so that the
+    ///                state is correct after the function returns
+    /// * `end_state` - The state to end in, this can either be `JtagState::RunTestIdle` or `JtagState::SelectDRScan`
+    fn shift_ir(
+        &self,
+        interface: &mut dyn DapProbe,
+        ir: u64,
+        state: &mut JtagState,
+        end_state: JtagState,
+    ) -> Result<(), ArmError> {
+        // THis is a wrapper around shift_reg that loads the IR register
+        self.shift_reg(
```

Comment:
This could be probably made even more generic with a proper implementation of the JTAG state machine ala openocd's `pathmove` where you just give a list of the states you want to visit 

[openocd commands](https://openocd.org/doc-release/html/JTAG-Commands.html)

Reply:
Yeah, sounds like an amazing plan :)
Happy to revisit later :)